spec/requests/maintenance_spec.rb

```diff
@@ -0,0 +1,20 @@
+require 'rails_helper'
+
+describe '[Maintenance mode]', type: :request do
+  it 'Maintenance mode can be switched on and off' do
+    login(dont_handle_splash: true)
```

Comment:
For this test, you can skip logging in altogether. You can just load the splash page directly (/welcome)